src/backend/distributed/commands/cascade_table_operation_for_connected_relations.c

```diff
@@ -342,45 +337,36 @@ ExecuteCascadeOperationForRelationIdList(List *relationIdList,
 	Oid relationId = InvalidOid;
 	foreach_oid(relationId, relationIdList)
 	{
-		CascadeOperationFunction cascadeOperationFunction =
-			GetCascadeOperationFunction(cascadeOperationType);
-
 		/*
 		 * Caller already passed the relations that we should operate on,
 		 * so we should not cascade here.
 		 */
-		bool cascadeViaForeignKeys = false;
-		cascadeOperationFunction(relationId, cascadeViaForeignKeys);
-	}
-}
-
-
-/*
- * GetCascadeOperationFunction returns c api for citus table operation according
- * to given CascadeOperationType.
- */
-static CascadeOperationFunction
-GetCascadeOperationFunction(CascadeOperationType cascadeOperationType)
-{
-	switch (cascadeOperationType)
-	{
-		case UNDISTRIBUTE_TABLE:
-		{
-			return UndistributeTable;
-		}
-
-		case CREATE_CITUS_LOCAL_TABLE:
-		{
-			return CreateCitusLocalTable;
-		}
-
-		default:
+		switch (cascadeOperationType)
 		{
-			/*
-			 * This is not expected as other create table functions don't have
-			 * cascade option yet. To be on the safe side, error out here.
-			 */
-			ereport(ERROR, (errmsg("citus table function could not be found")));
+			case CASCADE_FKEY_UNDISTRIBUTE_TABLE:
+			{
+				TableConversionParameters params = {
+					.relationId = relationId,
+					.cascadeViaForeignKeys = false
+				};
+				UndistributeTable(&params);
+				break;
+			}
+
+			case CREATE_CITUS_LOCAL_TABLE:
```

Comment:
can we rename this to something like `CASCADE_FKEY_CREATE_CITUS_LOCAL_TABLE` as we do CASCADE_FKEY_UNDISTRIBUTE_TABLE ?